src/generator/models/declaration.ts

```diff
@@ -166,6 +166,11 @@ function renderTypeScriptDeclarationForField({
        * The documentation of this field.
        */
       description: ${field.documentation ? `string` : `undefined`}
+
+      /**
+       * The resolver of this field
+       */
+      resolve: ${renderNexusResolve(model, field)}
```

Comment:
TL;DR you can ignore this comment

I would like to use the `resolve(...): ...` syntax as that affects how VSCode will present the intellisense autocomplete icon. By using `resolve:` VSCode will present this as a property, but `resolve(...): ...` will make it presented as a method. That is nicer because it tells you at a glance that it is not just a value but a function.

That said, I actually don't know how we'll achieve this with `NexusCore.FieldResolver` so probably just ignore this. Maybe just leave a TODO comment, I would like to leave an issue with the TS and/or VSCode team about being able to affect this without having to change syntax style, since e.g. in this case its totally not trivial to do so since we're working with type helpers, not manually writing from scratch.

MAYBE you can play around with `ReturnType`and Parameter extraction type helpers + spread syntax to reconstruct the method signature... however even if it works out it might lead to terrible looking types.

Probably what would have to happen is we pre-compute a simple type at build time, and for that to happen I think we'd open a can of worms as Nexus isn't optimized for that at all right now.